### PR TITLE
Gas syntax

### PR DESCRIPTION
**Pull request by danlrobertson**
I added this as I personally need syntax highlighting for gas assembly files for viewing disassembled C/C++/Rust code. As a result, I definitely will not be offended if you do not merge this pull request, especially since there are so many different assemblers. Comments and suggestions are welcome. I havn't come even close to covering the whole of gas assembly with this, but I plan to keep working on this, as I personally need this feature. I will need to write a kak file for LLVM IR in the near future as well, so let me know if you guys will want that once I get the ball rolling there.

I only just built and started using kakoune a few hours ago, so I'm a super noobie :smile:

Any suggestions on places in the code to look/explore as a noobie contributor are very very welcome! As a tmux + vim user/addict I dig this project!

### CODE REVIEW

File: rc/gas.kak [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Detection
+# ---------
+hook global BufSetOption mimetype=text/x-python %{
+    set buffer filetype gas
```
Comment (lenormf) [inline rc/gas.kak]: This is going to conflict with regular python files I reckon, not good.

File: rc/gas.kak [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+hook global BufCreate .*\.(s|S|asm)$ %{
+    set buffer filetype gas
```
Comment (lenormf) [inline rc/gas.kak]: This is could be a solution, except not all `.s` files use follow the GAS syntax. Some trickery might be needed here, maybe a quick grep into the buffer to find unique keywords that specific to this syntax, to be sure of the type of file you're dealing with ?

**Comment by lenormf** [comment]
On a side note: in my opinion the `rc` directory of the project should only contain files that support common languages, support for a particular flavor of ASM should go onto the [community scripts page](http://kakoune.org/scripts.html) (PRs on the `gh-pages` branch).

**Comment by mawww** [comment]
I agree we should avoid too exotic languages in rc/ however gas  is basically AT&T assembly syntax, which is the defacto standard assembly syntax on Linux, so I would be interested in merging that. Mime type detection should be fixed though, I dont know other filetypes that uses s/S as a common extension, so for the time being, if the file shell command does not give meaningful mimetype, the mimetype hook could just be removed.

**Comment by danlrobertson** [comment]
Sorry for the pythin mixup. I must have forgot to remove that after I copied it from the python file. My bad...

I'll remove that and I'll remove the mime type detection.

Since in intel assembly the registers are not prefixed with a % that could be an easy solution using grep. Thanks for the comments. I'll push some edits in a little bit.

**Comment by danlrobertson** [comment]
Removed the mime type detection and should have all of the general purpose instructions for x86 att/gas. I'll be adding floating point instructions sometime soon. Let me know if you see anything else that should be edited.

**Comment by lenormf** [comment]
@mawww OK then. How about a cleanup of the current list of scripts that's automatically loaded ? We can move some of the ones that are -arguably- not used much at all to the community list. A couple ones off the top of my head:
* coffeescript
* cucumber
* exherbo
* kickstart
* …

**Pull request merged**